### DISCUSSION TITLE
Displaying Current Language in Panel

### DISCUSSION THREAD

**Discussion by mr-scrpt** [FelixKratz/SketchyBar#345] [Q&A] [answered]
Hello. Could you please tell me how to display the current language in Sketchybar? For example, I have multiple keyboard layouts - English, Spanish, and German, and in the standard macOS panel, I can see which keyboard language is selected. I would like to display the selected language in Sketchybar as well.

**Comment by FelixKratz** [comment, marked as answer]
The easiest way to get this displayed in sketchybar is to alias the original macOS menu item into sketchybar:
```bash
sketchybar --add alias TextInputMenuAgent right
```
This should be all you need to do to get the input language shown in sketchybar just like it is in the regular bar.
(This is the name of the item on my system at least, you can check if it is the same on your system by running: `sketchybar --query default_menu_items`)


If you howerver want to customize the behavior and make a custom item, there is an event that can be used to detect changes in the input source i.e.:
```bash
sketchybar --add event input_change AppleSelectedInputSourcesChangedNotification
```
you could then use a regular item with a script and subscribe it to the above event for all the customization you want to do.

**Reply by zihao-liu-qs** [reply]
Thank U!!!
